src/js/script.js

```diff
@@ -0,0 +1,374 @@
+    } else if (ru && 'ru_' in key) {
+      letter = key.ru_;
+    } else if (shift && 'sh' in key) {
+      letter = key.sh;
+    } else {
+      letter = key.en.toUpperCase();
+    }
+  } else if (capsLock) {
+    letter = ru && 'ru' in key ? key.ru.toUpperCase() : key.en.toUpperCase();
+  } else {
+    letter = ru && 'ru' in key ? key.ru : key.en;
+  }
+  return letter;
+}
+
+function view() {
+  KEYS.forEach(row => {
+    row.forEach(key => {
+      if (key.form === 'k') {
+        let button = document.querySelector('#' + key.code);
+        button.textContent = changeOneKey(key);
+      }
+    });
+  });
+}
+
+function insert(str, pos, add) {
+  let arr = str.split('');
+  arr.splice(pos, 0, add);
+  return arr.join('');
+}
+
+function del(str, pos, after) {
+  let arr = str.split('');
+  if (after) {
+    arr.splice(pos, 1);
+  } else {
+    pos && arr.splice(pos - 1, 1);
+  }
+  return arr.join('');
+}
+
+function clearTimes(nameKey) {
+  delete times[nameKey];
+}
+
+function systemKey(keyCode) {
+  return (
+    keyCode === 'ShiftLeft' ||
+    keyCode === 'ShiftRight' ||
+    keyCode === 'AltLeft' ||
+    keyCode === 'AltRight' ||
+    keyCode === 'ControlLeft' ||
+    keyCode === 'ControlRight'
+  );
+}
+
+function deActiveSystem(code) {
+  if (code === 'ShiftLeft' || code === 'ShiftRight') {
+    shift = false;
+    view();
+  } else if (code === 'AltLeft' || code === 'AltRight') {
+    alt = false;
+  } else if (code === 'ControlLeft' || code === 'ControlRight') {
+    ctrl = false;
+  }
+}
+
+function clickKey(event) {
+  const TIMER_S = 1000;
+  const TIMER_K = 100;
+  let now = new Date().getTime();
+  let element = event.target;
+  if (!element.parentElement.classList.contains('outer')) {
+    return;
+  }
+  let keyCode = element.id;
+  if (event.type === 'mouseup') {
+    let interval = now - times[keyCode];
+    if (systemKey(keyCode) && interval < TIMER_S) {
+      setTimeout(() => {
+        element.classList.remove('active');
+        clearTimes(keyCode);
+        deActiveSystem(keyCode);
+      }, TIMER_S);
+    } else if (!systemKey(keyCode) && interval < TIMER_K) {
+      setTimeout(() => {
+        element.classList.remove('active');
+        clearTimes(keyCode);
+      }, TIMER_K);
+    } else {
+      element.classList.remove('active');
+      clearTimes(keyCode);
+      deActiveSystem(keyCode);
+    }
+    textarea.focus();
+    return;
+  }
+  if (times[keyCode] && now - times[keyCode] < TIMER_K) {
+    return;
+  }
+  element.classList.add('active');
+  elementKey(element);
+}
+
+function lang() {
+  ru = !ru;
+  if (ru) {
+    localStorage['ru'] = true;
+  } else {
+    localStorage.clear();
+  }
+  view();
+}
+
+function elementKey(el) {
+  let found = el.id;
+  let parent = el.parentElement;
+  let pos = textarea.selectionStart;
+  if (parent.classList.contains('key')) {
+    textarea.value = insert(textarea.value, pos, el.textContent);
+    pos++;
+  } else if (el.id === 'Space') {
```

Comment:
The same here, would be better to use 'switch' operator when you have to compare the same value with different variants.